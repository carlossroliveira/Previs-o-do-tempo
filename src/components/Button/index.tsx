// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ButtonSC } from './buttonStyles';

export const Button = (): JSX.Element => {
  return (
    <ContainerSC>
      <ButtonSC>Buscar</ButtonSC>
    </ContainerSC>
  );
};
