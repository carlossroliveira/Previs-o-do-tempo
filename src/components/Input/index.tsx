// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useMemo } from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { useMyHookApplication } from '../../Context/contextApplication/ContextTheme';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, InputSC } from './inputStyles';
import { Hourglass } from 'react-spinners-css';

export const Input = (): JSX.Element => {
  const { storage, loading, validateInput, onChangeInformation } =
    useMyHookApplication();

  const errorMessage = useMemo(() => {
    return !!validateInput;
  }, [validateInput]);

  return (
    <ContainerSC error={errorMessage}>
      <InputSC
        type="text"
        placeholder="Digite seu estado"
        value={storage}
        onChange={onChangeInformation}
      />

      {loading && <Hourglass color={'#393E46'} />}
    </ContainerSC>
  );
};
