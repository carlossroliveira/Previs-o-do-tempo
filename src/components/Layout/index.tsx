// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Header } from './part/Header';
import { ContentOne } from './part/ContentOne';
import { ContentTwo } from './part/ContentTwo';
import { Footer } from './part/Footer';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './layoutStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Layout = (): JSX.Element => {
  return (
    <ContainerSC>
      <Header />
      <ContentOne />
      <ContentTwo />
      <Footer />
    </ContainerSC>
  );
};
