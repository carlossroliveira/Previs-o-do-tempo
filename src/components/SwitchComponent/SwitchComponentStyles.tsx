import styled from 'styled-components';
import Switch, { ReactSwitchProps } from 'react-switch';

export const Toggle = styled(Switch).attrs<ReactSwitchProps>(({ theme }) => ({
  onColor: theme.background.secondary,
  offColor: theme.background.secondary,
  onHandleColor: theme.background.primary,
}))<ReactSwitchProps>``;

export const ParagraphSC = styled.p`
  margin-bottom: 3rem;
  font-weight: 600;
  font-style: italic;
`;
