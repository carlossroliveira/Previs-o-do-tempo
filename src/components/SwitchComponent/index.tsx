// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useState } from 'react';
import { useMyHook } from '../hooks/theme';
import { ParagraphSC, Toggle } from './SwitchComponentStyles';

export const SwitchComponent = (): JSX.Element => {
  const { handleThemes } = useMyHook();

  const [first, setfirst] = useState<boolean>(false);

  const test = () => {
    setfirst((firstt) => !firstt);
    handleThemes();
  };

  return (
    <>
      <ParagraphSC>{!first ? 'Light' : 'Dark'}</ParagraphSC>
      <Toggle
        checked={first}
        uncheckedIcon={false}
        checkedIcon={false}
        onChange={test}
      />
    </>
  );
};
