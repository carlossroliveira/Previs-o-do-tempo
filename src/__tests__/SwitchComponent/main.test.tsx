// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import theme from '../../styles/themes/dark';
import { SwitchComponent } from '../../components/SwitchComponent';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../utils/withThemeComponent';

const SwitchComponentWithTheme = WithThemeComponent(SwitchComponent);

describe('Switch Testing', () => {
  it('Should render the Switch component with name Dark', () => {
    render(<SwitchComponentWithTheme />);
    expect(screen.getByText(/dark/i)).toBeInTheDocument();
  });

  it('should render the Switch component enabled', () => {
    render(<SwitchComponentWithTheme />);
    expect(screen.getByText(/dark/i)).not.toBeDisabled();
  });

  it('should render the Switch component "not" with the right background selected', () => {
    render(<SwitchComponentWithTheme />);
    expect(screen.getByText(/dark/i)).not.toHaveStyle({
      'background-color': theme.background.primary,
    });
  });

  it('should match snapshot', () => {
    const component = render(<SwitchComponentWithTheme />);
    expect(component).toMatchSnapshot();
  });
});
