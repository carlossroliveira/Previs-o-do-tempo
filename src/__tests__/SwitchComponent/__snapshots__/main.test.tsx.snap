// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Testing should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  margin-bottom: 3rem;
  font-weight: 600;
  font-style: italic;
}

<body>
    <div>
      <p
        class="c0"
      >
        Dark
      </p>
      <div
        class=""
        style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
      >
        <div
          class="react-switch-bg"
          style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(238, 238, 238); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
        />
        <div
          class="react-switch-handle"
          style="height: 26px; width: 26px; background: rgb(57, 62, 70); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
        />
        <input
          aria-checked="false"
          role="switch"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
          type="checkbox"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  margin-bottom: 3rem;
  font-weight: 600;
  font-style: italic;
}

<div>
    <p
      class="c0"
    >
      Dark
    </p>
    <div
      class=""
      style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 14px; transition: opacity 0.25s; user-select: none;"
    >
      <div
        class="react-switch-bg"
        style="height: 28px; width: 56px; margin: 0px; position: relative; background: rgb(238, 238, 238); border-radius: 14px; cursor: pointer; transition: background 0.25s;"
      />
      <div
        class="react-switch-handle"
        style="height: 26px; width: 26px; background: rgb(57, 62, 70); display: inline-block; cursor: pointer; border-radius: 50%; position: absolute; transform: translateX(1px); top: 1px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
      />
      <input
        aria-checked="false"
        role="switch"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
        type="checkbox"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
